freediscovery/interpretation.py

```diff
@@ -0,0 +1,170 @@
+# -*- coding: utf-8 -*-
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+import re
+import matplotlib as mpl
+import matplotlib.cm as cm
+import numpy as np
+
+
+def explain_logistic_regression(weights, textlines):
+    """
+    Generate decorated HTML by adding <span> tags with background color to keywords depending on word's weight.
+
+    Parameters
+    ----------
+    weights: [dict] {vocabulary : weights} with words [str] as keys, weights (scores) [float] as values
+    textlines [list of str] list of text lines representing raw document text
+
+    Returns
+    -------
+    [str] HTML string representing document's text decorated with <span> tags
+    """
+    document_html = ''
+    for line in textlines:
+        line = line.replace('<', '',).replace('>', '')  # in order to have a valid HTML output after adding <span> tags
+        line_html = _replace_with_color_spans(line, weights)
+        document_html += line_html + '<br/>'

```

Comment:
The `+` operator [is quite inefficient on strings in a loop](https://stackoverflow.com/a/10043957/1791279), let's rather do `document_html = []`, then append strings to it and return a `"".join(document_html)`